ucp/_libs/ucp_py.pyx

```diff
@@ -38,6 +41,20 @@ PENDING_MESSAGES = {}  # type: Dict[Message, Future]
 UCX_FILE_DESCRIPTOR = -1
 reader_added = 0
 UCP_INITIALIZED = False
+LOGGER = None
+
+def ucp_logger(fxn):
+    def wrapper(*args, **kwargs):
+        global LOGGER
+        args_repr = [repr(a) for a in args]
+        kwargs_repr = [f"{k}={v!r}" for k, v in kwargs.items()]
+        signature = ", ".join(args_repr + kwargs_repr)
+        LOGGER.debug(f"Calling {fxn.__name__}({signature})")
+        rval = fxn(*args, **kwargs)
+        LOGGER.debug(f"{fxn.__name__!r} returned {rval!r}")
```

Comment:
FYI, this isn't free.  In Dask we ended up removing debug calls from within our inner loops because they took up non-trivial time.  This shouldn't matter unless you're above the 10k msgs/s rate (a guess), but I thought I'd mention it in case you care about that level.

Reply:
@mrocklin I've made the logger a no-op if the `UCXPY_LOG_LEVEL` env var isn't set.